app/src/main/java/com/nervousfish/nervousfish/activities/MainActivity.java

```diff
@@ -173,16 +189,18 @@ public void onPairingButtonClicked(final View view) {
                 break;
             case R.id.pairing_menu_nfc:
                 intent.setComponent(new ComponentName(this, NFCActivity.class));
+                intent.putExtra(ConstantKeywords.SERVICE_LOCATOR, this.serviceLocator);
+                this.startActivity(intent);
                 break;
             case R.id.pairing_menu_qr:
                 intent.setComponent(new ComponentName(this, QRExchangeKeyActivity.class));
+                intent.putExtra(ConstantKeywords.SERVICE_LOCATOR, this.serviceLocator);
+                this.startActivity(intent);
                 break;
             default:
                 LOGGER.error("Unknown pairing button clicked");
                 throw new IllegalArgumentException("Only existing buttons can be clicked");
         }
-
-        this.startActivity(intent);
```

Comment:
You can return early after displaying the popup (which is already the case, see [line 187](https://github.com/ericcornelissen/NervousFish/pull/195/files/8b94f909f6d413b81431fd77d2182ed44556b53c#diff-149538cc1f1f5e4370b6ae93d6489330R187)) 😉